libyul/optimiser/ForLoopConditionIntoBody.h

```diff
@@ -48,4 +48,35 @@ class ForLoopConditionIntoBody: public ASTModifier
 	Dialect const& m_dialect;
 };
 
+/**
+ * Reverses the transformation of ForLoopConditionIntoBody.
+ *
+ * for { ... } 1 { ... } {
+ *   if iszero(c) { break }
+ *   ...
+ * }
+ *
+ * will be turned into
+ *
+ * for { ... } c { ... } {
+ *   ...
+ * }
+ *
+ * if ``c`` is movable.
+ *
+ * The LiteralRematerialiser should be run before this step.
+ */
+class ForLoopConditionOutOfBody: public ASTModifier
```

Comment:
Ok, it's small and closely related to the other one, but might it still deserve it's own file :-)?